examples/02_explaining_similarity_encoding.py

```diff
@@ -0,0 +1,148 @@
+works, and how it compares to the |OneHotEncoder| and other encoders in the
+library.
+
+
+.. |OneHotEncoder| replace::
+    :class:`~sklearn.preprocessing.OneHotEncoder`
+
+.. |SimilarityEncoder| replace::
+    :class:`~skrub.SimilarityEncoder`
+"""
+
+###############################################################################
+# What do we mean by dirty categories?
+# ------------------------------------
+#
+# Let's get the table we will use in this example,
+# the `employee salaries dataset <https://www.openml.org/d/42125>`_:
+
+from skrub.datasets import fetch_employee_salaries
+
+dataset = fetch_employee_salaries()
+print(dataset.description)
+
+X = dataset.X
+y = dataset.y
+
+X.head()
+
+
+###############################################################################
+# These different entries are often variations of the same entity.
+# For example, there are 3 kinds of "Accountant/Auditor".
+#
+# Such variations will break traditional categorical encoding methods:
+#
+# * Using a simple |OneHotEncoder|
+#   will create orthogonal features, whereas it is clear that
+#   those 3 terms have a lot in common.
+#
+# * If we wanted to use word embedding methods such as
+#   `Word2vec <https://www.tensorflow.org/tutorials/text/word2vec>`_,
+#   we would have to go through a cleaning phase: those algorithms
+#   are not trained to work on data such as "Accountant/Auditor I".
+#   However, this can be error-prone and time-consuming.
+#
+# The problem becomes easier if we can capture relationships between
+# entries.
+#
+# To simplify understanding, we will focus on the column describing the
+# employee's position title:
+
+values = X[["employee_position_title", "gender"]]
+values.insert(0, "current_annual_salary", dataset.y)
+
+sorted_values = values["employee_position_title"].sort_values().unique()
+
+###############################################################################
+# In order to robustly embed dirty semantic data, the |SimilarityEncoder|
+# creates a similarity matrix based on an n-gram representation of the data.
+
+from skrub import SimilarityEncoder
+
+similarity_encoder = SimilarityEncoder()
+transformed_values = similarity_encoder.fit_transform(sorted_values.reshape(-1, 1))
+
+###############################################################################
+# Plotting the new representation using multi-dimensional scaling
+# ---------------------------------------------------------------
+#
+# Let's now plot a couple of points at random using a low-dimensional
+# representation to get an intuition of what the |SimilarityEncoder| is doing:
+
+from sklearn.manifold import MDS
+
+mds = MDS(dissimilarity="precomputed", n_init=10, random_state=42)
+two_dim_data = mds.fit_transform(1 - transformed_values)
+# transformed values lie in the 0-1 range,
+# so 1-transformed_value yields a positive dissimilarity matrix
+print(two_dim_data.shape)
+print(sorted_values.shape)
```

Comment:
If you insert the above explanations, I feel you don't need this, and you can directly do the plot with the points.